stdlib/LinearAlgebra/src/matmul.jl

```diff
@@ -1081,3 +1081,157 @@ function matmul3x3!(C::AbstractMatrix, tA, tB, A::AbstractMatrix, B::AbstractMat
     end # inbounds
     C
 end
+
+const RealOrComplex = Union{Real,Complex}
+
+# Three-argument *
+"""
+    *(A, B::AbstractMatrix, C)
+    A * B * C * D
+
+Chained multiplication of 3 or 4 matrices is done in the most efficient sequence,
+based on the sizes of the arrays. That is, the number of scalar multiplications needed
+for `(A * B) * C` (with 3 dense matrices) is compared to that for `A * (B * C)`
+to choose which of these to execute.
+
+If the last factor is a vector, or the first a transposed vector, then it is efficient
+to deal with these first. In particular `x' * B * y` means `(x' * B) * y`
+for an ordinary colum-major `B::Matrix`. This is often equivalent to `dot(x, B, y)`,
+although it allocates an intermediate array.
+
+If the first or last factor is a number, this will be fused with the matrix
+multiplication, using 5-arg [`mul!`](@ref).
+
+See also [`muladd`](@ref), [`dot`](@ref).
+
+!!! compat "Julia 1.7"
+    These optimisations require at least Julia 1.7.
+
+# Examples
+```
+julia> A, B, C = randn(100,10), randn(10,100), randn(100,10);
+
+julia> using BenchmarkTools
+
+julia> @btime ($A * $B) * $C;  # slow
+  13.500 μs (3 allocations: 86.14 KiB)
```

Comment:
My dollars are broken, but before I escape them... is this a good idea, or just clutter?

Reply:
I'm not sure this adds anything to the description above. Personally, I'd suggest to remove the examples part.

Reply:
OK, it's gone.